source/assets/js/playground/autocomplete.ts

```diff
@@ -0,0 +1,241 @@
+  const atRule = context.matchBefore(/@\w*/);
+  if (!atRule) return null;
+  if (atRule.from === atRule.to && !context.explicit) return null;
+  return {
+    from: atRule.from,
+    to: atRule.to,
+    options: atRuleOptions,
+    validFor: identifier,
+  };
+}
+
+type CompletionInfo = {
+  name: string;
+  description?: string;
+};
+
+type CompletionModule = CompletionInfo & {
+  functions?: CompletionInfo[];
+  variables?: CompletionInfo[];
+};
+
+const moduleNames = moduleMembers.map(mod => mod.name);
+
+const moduleDescriptions = {
+  color:
+    'generates new colors based on existing ones, making it easy to build color themes',
+  list: 'lets you access and modify values in lists',
+  map: 'makes it possible to look up the value associated with a key in a map, and much more',
+  math: 'provides functions that operate on numbers',
+  meta: 'exposes the details of Sass’s inner workings',
+  selector: 'provides access to Sass’s powerful selector engine',
+  string: 'makes it easy to combine, search, or split apart strings',
+};
+
+const builtinModules: CompletionModule[] = moduleMembers.map(modMember => {
+  let variables;
+  if ('variables' in modMember) {
+    variables = modMember.variables;
+  }
+  return {
+    name: modMember.name,
+    description: moduleDescriptions[modMember.name],
+    functions: (modMember.functions ?? []).map(name => ({name})),
+    variables: (variables ?? []).map(name => ({name: `$${name}`})),
+  };
+});
+
+const moduleNameRegExp = new RegExp(`(${moduleNames.join('|')}).\\$?\\w*`);
+
+const moduleCompletions = Object.freeze(
+  builtinModules.map(mod => ({
+    label: `sass:${mod.name}`,
+    apply: `sass:${mod.name}`,
+    info: mod.description,
+    type: 'class',
+    validFor: identifier,
+  }))
+);
+
+// Completions for the import of built in modules, for instance "sass:color".
+function builtinModulesCompletion(
+  context: CompletionContext
+): CompletionResult | null {
+  const nodeBefore = syntaxTree(context.state).resolveInner(context.pos, -1);
+  const potentialTypes = [
+    'StringLiteral', // When wrapped in quotes- `"sass:"`
+    'ValueName', // No end quote, before semicolon- `"sa`
+    ':', // No end quote, on semicolon- `"sass:`
+    'PseudoClassName', // No end quote, after semicolon- `"sass:m`
+  ];
+  if (!potentialTypes.includes(nodeBefore.type.name)) return null;
+  const potentialParentTypes = [
+    'UseStatement', // When wrapped in quotes
+    'PseudoClassSelector', // No end quote, after the colon
+  ];
+  if (!potentialParentTypes.includes(nodeBefore.parent?.type.name || ''))
+    return null;
+
+  const moduleMatch = context.matchBefore(/["'](sass:)?\w*/);
+
+  if (!moduleMatch) return null;
+  if (moduleMatch.from === moduleMatch.to && !context.explicit) return null;
+  return {
+    from: moduleMatch.from + 1,
+    to: moduleMatch.to,
+    options: moduleCompletions,
+    validFor: identifier,
+  };
+}
+
+const moduleVariableCompletions = Object.freeze(
```

Comment:
Probably for a follow-up PR, but it's worth thinking about ways to extract the documentation for variables and functions from `source/documentation/modules/*.md` so we can include them in these autocompletions.